Services/GEV/Utils/classes/class.gevOrgUnitUtils.php

```diff
@@ -995,18 +995,45 @@ public static function getSuperiorsOfUser($user_id) {
 		return gevUserUtils::removeInactiveUsers($sups);
 	}
 
-	static public function getOrguUnitsXLevelAbove($org_ref_id, $lvl) {
+	static public function getBDOf($org_ref_id) {
+		require_once("Services/GEV/Utils/classes/class.gevSettings.php");
+		require_once("Services/GEV/Utils/classes/class.gevObjectUtils.php");
 		$ou_tree = ilObjOrgUnitTree::_getInstance();
-		$ret = $org_ref_id;
-		for($i = 0; $i< $lvl; $i++) {
-			$ret = $ou_tree->getParent($ret);
 
-			if($ret == ilObjOrgUnit::getRootOrgRefId()) {
-				break;
+		$parent = $ou_tree->getParent($org_ref_id);
+		if(self::isBD(ilObject::_lookupObjId($parent))) {
+			return $parent;
+		}
+
+		return gevOrgUnitUtils::getBDOf($parent);
+	}
+
+	static public function getBDByName($bd_name, $org_ref_id = null) {
```

Comment:
Sie sucht die BD Anhand des Namens. Gab es so auch schon früher, war da nur doppelt und ich hab sie zentralisiert.
Der zweite Parameter ermöglicht den Einstieg in die Suche an einer andere Stelle außer der UVG Root Organisationseinheit. Ist mit eingebaut, falls die Generalis sich mal unter der UVG nochmal eine Unterteilung wünscht, bevor dann die BDen kommen.